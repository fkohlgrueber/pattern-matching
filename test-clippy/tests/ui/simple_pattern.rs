

fn main() {
    
    // match literal false
    if (false) {
    }

    // match array sequences
    let a = ['b', 'c'];
    let b = ['b', 'b', 'c'];
    let no_warn = ['b'];
    
    // match cast
    let x = 0 as *const i32;    

    // match if
    let y = if true {
        123
    } else {
        456
    };
    if true {
        true;
    }
    let z = if true {
        true;
        12
    }else {
        34
    };
}
