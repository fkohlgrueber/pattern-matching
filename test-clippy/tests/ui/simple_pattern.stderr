error: This is a match for a simple pattern. Well Done!
  --> $DIR/simple_pattern.rs:6:9
   |
LL |     if (false) {
   |         ^^^^^
   |
   = note: #[forbid(simple_pattern)] on by default

error: This is a match for a simple pattern. Well Done!
  --> $DIR/simple_pattern.rs:10:13
   |
LL |     let a = ['b', 'c'];
   |             ^^^^^^^^^^

error: This is a match for a simple pattern. Well Done!
  --> $DIR/simple_pattern.rs:11:13
   |
LL |     let b = ['b', 'b', 'c'];
   |             ^^^^^^^^^^^^^^^

error: This is a match for a simple pattern. Well Done!
  --> $DIR/simple_pattern.rs:18:13
   |
LL |       let y = if true {
   |  _____________^
LL | |         123
LL | |     } else {
LL | |         456
LL | |     };
   | |_____^

error: This is a match for a simple pattern. Well Done!
  --> $DIR/simple_pattern.rs:23:5
   |
LL | /     if true {
LL | |         true;
LL | |     }
   | |_____^

error: aborting due to 5 previous errors

